docs/iota-did-method-spec.md

```diff
@@ -0,0 +1,128 @@
+# IOTA DID Method Specification
+
+Version 0.1-draft by Jelle Millenaar, IOTA Foundation
+
+## Abstract
+
+The IOTA DID Method Specification describes a method of implementing the [Decentralized Identifiers](https://www.w3.org/TR/did-core/) (DID) standard on the IOTA Tangle, a Distributed Ledger Technology (DLT). It conforms to the [DID specifications v1.0 Working Draft 20200731](https://www.w3.org/TR/2020/WD-did-core-20200731/) and described how to publish DID Document Create, Read, Update and Delete (CRUD) operations to the IOTA Tangle. 
+
+## Introduction
+
+### The IOTA Tangle
+
+This specification defines a method of implementing DID on top of the IOTA Tangle, which is a Distributed Ledger Technology (DLT) using a Tangle data structure. In contrast to a Blockchain, the Tangle does not store messages in blocks and chain them together, but rather creates a data structure where a message references two previous messages, creating a parallel structure. 
+
+For this method, the most important features of IOTA are: 
+
+* The lack of fees, requiring no cryptocurrency tokens to be owned in order to submit a message to the DLT.
+* Pure data messages are possible to be stored immutably. 
+* Few nodes store the entire Tangle, requiring additional logic to prove the immutability of data. 
+
+## DID Method Name
+
+The namestring to identify this DID method is: `iota`.
+
+A DID that uses this method MUST begin with the following prefix: `did:iota`. Following the generic DID specification, this string MUST be completely in lowercase.
+
+## DID Format
+
+The DIDs that follow this method have the following format:
+```
+iota-did = "did:iota:" iota-specific-idstring
+iota-specific-idstring = [ iota-network ":" ] [ network-shard ":" ] iota-tag
+iota-network = 6* char
+network-shard = 6* char
+iota-tag = 44* base-char
```

Comment:
The length can be 43 or 44 chars

Reply:
Why either? It should be one or the other, right?

Reply:
I just checked it again, the actual length can be even smaller (32), if all bytes are 0
During the base58 conversion leading zeros are replaced by a 1, so `[ 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]` will become 11111111111111111111111111111111 and `[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 2, 3, 4, 5, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]`will become 111111111111113RdG934jKrkSo4czdsWewTm1

Reply:
Can the minimum length just be `32`?